src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipVerifiersTest.java

```diff
@@ -82,4 +88,32 @@ public void should_not_verify_invalid_zip_successfully() throws Exception {
         ZipVerifiers.sha256WithRsaVerification(zipStreamWithSig);
     }
 
+    @Test
+    public void should_verify_valid_xyz_test_zip_successfully() throws Exception {
+        byte[] test1XyzBytes = toByteArray(getResource("signature/xyz_test_envelope.zip"));
+        byte[] test1SigXyzBytes = toByteArray(getResource("signature/xyz_test_signature"));
+        assertThat(ZipVerifiers.verifySignature(xyzPublicKeyBase64, test1XyzBytes, test1SigXyzBytes)).isTrue();
+    }
+
+    @Test
+    public void should_not_verify_invalid_xyz_test_zip_successfully() throws Exception {
+        byte[] test1XyzBytes = toByteArray(getResource("signature/envelope.zip"));
+        byte[] test1SigXyzBytes = toByteArray(getResource("signature/xyz_test_signature"));
+        assertThat(ZipVerifiers.verifySignature(xyzPublicKeyBase64, test1XyzBytes, test1SigXyzBytes)).isFalse();
+    }
+
+    @Test
+    public void should_not_verify_invalid_xyz_signature_successfully() throws Exception {
+        byte[] test1XyzBytes = toByteArray(getResource("signature/xyz_test_envelope.zip"));
+        byte[] test1SigXyzBytes = toByteArray(getResource("signature/signature"));
+        assertThat(ZipVerifiers.verifySignature(xyzPublicKeyBase64, test1XyzBytes, test1SigXyzBytes)).isFalse();
+    }
+
+    @Test
+    public void should_not_verify_valid_xyz_test_zip_with_wrong_public_key_successfully() throws Exception {
+        byte[] test1XyzBytes = toByteArray(getResource("signature/xyz_test_envelope.zip"));
+        byte[] test1SigXyzBytes = toByteArray(getResource("signature/xyz_test_signature"));
```

Comment:
I would think based on file names that this is the correct signature for the file above

Reply:
The signature is in fact correct, but the public key is not as it says on the "tin"